sqlglot/optimizer/qualify_columns.py

```diff
@@ -0,0 +1,321 @@
+import sqlglot.expressions as exp
+
+
+def qualify_columns(expression, schema):
+    """
+    Rewrite sqlglot AST to have fully qualified columns.
+
+    Example:
+        >>> import sqlglot
+        >>> schema = {"tbl": {"col": "INT"}}
+        >>> expression = sqlglot.parse_one("SELECT col FROM tbl")
+        >>> qualify_columns(expression, schema).sql()
+        'SELECT tbl.col AS col FROM tbl'
+
+    Args:
+        expression (sqlglot.Expression): expression to qualify
+        schema (dict): Mapping of table names to all available columns
+    Returns:
+        sqlglot.Expression: qualified expression
+    """
+    expression = expression.copy()
+    _qualify_union(expression, schema, Sequence(), {})
+    return expression
+
+
+class Sequence:
```

Comment:
maybe this should go in helpers

Reply:
Actually, Python has a built in for this: https://docs.python.org/3/library/itertools.html#itertools.count
I'll just use that